specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json

```diff
@@ -0,0 +1,416 @@
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/natGateways": {
+      "get": {
+        "tags": [
+          "NatGateways"
+        ],
+        "operationId": "NatGateways_ListAll",
+        "description": "Gets all the Nat Gateways in a subscription.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of NatGateway resources.",
+            "schema": {
+              "$ref": "#/definitions/NatGatewayListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List all nat gateways": { "$ref": "./examples/NatGatewayListAll.json" }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/natGateways": {
+      "get": {
+        "tags": [
+          "NatGateways"
+        ],
+        "operationId": "NatGateways_List",
+        "description": "Gets all nat gateways in a resource group.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of NatGateway resources.",
+            "schema": {
+              "$ref": "#/definitions/NatGatewayListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List nat gateways in resource group": { "$ref": "./examples/NatGatewayList.json" }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "NatGatewaySku": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of Nat Gateway SKU.",
+          "enum": [
+            "Basic",
+            "Standard"
+          ],
+          "x-ms-enum": {
+            "name": "NatGatewaySkuName",
+            "modelAsString": true
+          }
+        }
+      },
+      "description": "SKU of nat gateway"
+    },
+    "NatGatewayPropertiesFormat": {
+      "properties": {
+        "idleTimeoutInMinutes": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The idle timeout of the nat gateway."
+        },
+        "publicIpAddresses": {
```

Comment:
please ensure linked access checks are defined for these referenced resourceids

Reply:
thanks, this is updated here  https://msazure.visualstudio.com/_search?text=natgateway&type=code&lp=custom-Collection&result=DefaultCollection%2FOne%2FNetworking-nrp%2FGBmaster%2F%2Fsrc%2Fconfigs%2FEnvironmentConfigs%2Fcsm_registration_production.json&filters=ProjectFilters%7BOne%7DRepositoryFilters%7BNetworking-nrp%7D&_a=contents